solidity/test/bridge/WalletCoordinator.test.ts

```diff
@@ -0,0 +1,921 @@
+            createdAt: 0,
+            movingFundsRequestedAt: 0,
+            closingStartedAt: 0,
+            pendingMovedFundsSweepRequestsCount: 0,
+            state: 1,
+            movingFundsTargetWalletsCommitmentHash: HashZero,
+          })
+
+          walletRegistry.isWalletMember
+            .whenCalledWith(
+              ecdsaWalletID,
+              walletMembersIDs,
+              thirdParty.address,
+              1
+            )
+            .returns(false)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+
+          bridge.wallets.reset()
+          walletRegistry.isWalletMember.reset()
+        })
+
+        it("should revert", async () => {
+          const tx = walletCoordinator
+            .connect(thirdParty)
+            .submitDepositSweepProposalWithReimbursement(
+              {
+                walletPubKeyHash,
+                depositsKeys: [
+                  {
+                    fundingTxHash:
+                      "0x51f373dcbb6122bcb1c62964b5f3be923092dc64bc9e31257931d58c4eadb9f5",
+                    fundingOutputIndex: 0,
+                  },
+                ],
+              },
+              {
+                walletMembersIDs,
+                walletMemberIndex: 1,
+              }
+            )
+
+          await expect(tx).to.be.revertedWith(
+            "Caller is neither a proposal submitter nor a wallet member"
+          )
+        })
+      }
+    )
+
+    // Here we just check that the reimbursement works. Detailed
+    // assertions are already done within the scenario stressing the
+    // `submitDepositSweepProposal` function.
+    context(
+      "when the caller is a proposal submitter or a wallet member",
+      () => {
+        before(async () => {
+          await createSnapshot()
+
+          reimbursementPool.refund.returns()
+
+          await walletCoordinator
+            .connect(owner)
+            .addProposalSubmitter(thirdParty.address)
+
+          await walletCoordinator
+            .connect(thirdParty)
+            .submitDepositSweepProposalWithReimbursement(
+              {
+                walletPubKeyHash,
+                depositsKeys: [
+                  {
+                    fundingTxHash:
+                      "0x51f373dcbb6122bcb1c62964b5f3be923092dc64bc9e31257931d58c4eadb9f5",
+                    fundingOutputIndex: 0,
+                  },
+                ],
+              },
+              emptyWalletMemberContext
+            )
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+
+          reimbursementPool.refund.reset()
+        })
+
+        it("should do the refund", async () => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          expect(reimbursementPool.refund).to.have.been.calledOnce
+
+          expect(reimbursementPool.refund.getCall(0).args[0]).to.be.closeTo(
+            BigNumber.from(32000),
+            2000
+          )
```

Comment:
This assertion is not actually checking if the caller is not net-negative, I am afraid.

How complicated would it be to implement tests with a real `ReimbursementPool`?

Reply:
I don't think using real pool changes anything in this context. After all, it's about calling `reimbursementPool.refund` with proper parameters.

Reply:
Right, and the *proper parameters* are the problem. 😉 See [this](https://github.com/keep-network/tbtc-v2/pull/575#discussion_r1159593044) comment - right now we are assuming no gas offset is needed. Either way, we should confirm it in a test - gas offset is not needed or gas offset is needed and the value is `x`.

Reply:
Yep yep, I'll reply in the mentioned comment.